hack/tools/Makefile

```diff
@@ -30,6 +30,7 @@ KBLD_VERSION=v0.31.0
 KAPP_VERSION=v0.42.0
 VENDIR_VERSION=v0.23.0
 VALE_VERSION=2.17.0
+GOLANGCI_LINT_VERSION=1.46.0
```

Comment:
Good question. I think I was seeing issues using the 1.43 version with go 1.18, so was anticipating we might need to bump soon. 
Let me retry (since I notice I need to update the go.mod more thoroughly) and see if I can keep it on 1.43 and bump the version later.

Reply:
yeah reverting back to 1.43 on m1 mac got this:

```
panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt

goroutine 1 [running]:
github.com/go-critic/go-critic/checkers.init.22()
        github.com/go-critic/go-critic@v0.6.2/checkers/embedded_rules.go:46 +0x52c
```
The issue seems specific to 1.18 (and arm64). Given there are at least some folks on 1.18 , it seems prudent to bump to a version that have been known to work.

I will be monitoring the linter outputs to see if the bump to 1.46 has any material effect on them.

Reply:
A few minor differences in the output which I have addressed as well.